File: Distribute_Candies_to_People.java
class Solution {
    public int[] distributeCandies(int candies, int num_people) {
        int[] ans = new int[num_people];
        int k=0,v=1;
        while(candies>0){
            if(k == num_people)
            k = k % num_people; //to loop through the array
            if(candies < v){
                ans[k] += candies; break;
            }
            ans[k++] += v;
            candies -= v++;
        }
        return ans;
    }
}

File: Distribute_Candies_to_People.cpp
class Solution {
public:
    vector<int> distributeCandies(int candies, int num_people) {
    vector<int> ans(num_people,0);
    int count=1;
    for(int i=0;i<=num_people && candies>0;i++){
        if(i==num_people){
            i=0;
        }
        int v=min(count,candies);
        ans[i]+=v;
        count++;
        candies-=v;
    }
    return ans;
    }
};
